.github/actions/validate-repo-version/action.yml

```diff
@@ -0,0 +1,49 @@
+name: Validate Repo Version
+description: Action that validates that a version in `config/versions.json` matches the version in the repository
+inputs:
+  repo-to-check:
+    type: string
+    required: true
+    description: ACCESS-NRI repository to validate associated version in `config/versions.json`
+outputs:
+  version:
+    value: ${{ steps.jq.outputs.version }}
+    description: Version of spack-packages from the `config/versions.json`
+runs:
+  using: composite
+  steps:
+    # Checkout the callers `config/versions.json`
+    - uses: actions/checkout@v4
```

Comment:
Is there a situation where we'd want to check this for a version of the model that is not the `HEAD` of `main`?

Reply:
Indeed - and that case is covered. `actions/checkout` will by default have a `ref` that is dependent on the event type. So for `on.pull_request` it will be the PR branch, and on others it will be `main`. Unless you mean user-customisable, in which case - no. But I don't know if it should be.

Reply:
I'm thinking about the scheduled check case for a configuration, but perhaps that doesn't intersect with this?

In which case merge away!

Reply:
No, we don't validate that because in order for it to have been merged and tagged at that point, it must have passed that check during PR